### DISCUSSION TITLE
Mobi or R : How to transfer values of all state variables of a system at steady state to the simulation

### DISCUSSION THREAD

**Discussion by Sulav1** [Open-Systems-Pharmacology/Forum#1450] [Questions & Problems]
Is there a way to tranfer values of all state variables of a simulation at steady state to the molecular start value? I intend to create a mobi simulation with reactions that can be at steady state. In order to do so,  I require to run the simulation for a certain time horizon such that the system is at steady state and then I need to manually  transfer the values of all variables to the molecular start value. I was wondering, if someone came across similar issue and found a generic solution using R to do so. @Yuri05  any comment or suggestions

**Comment by Yuri05** [comment]
@PavelBal Think you did it in R already and can provide an example script?

**Comment by PavelBal** [comment]
Hi Sulav,

yes we have such a function as part of the Parameter Identification package (https://github.com/Open-Systems-Pharmacology/OSPSuite.ParameterIdentification) that is currently being developed. Check out this file

https://github.com/Open-Systems-Pharmacology/OSPSuite.ParameterIdentification/blob/develop/R/utilities-simulation.R

This function is used in the esqlabsR package (https://github.com/esqLABS/esqlabsR) here https://github.com/esqLABS/esqlabsR/blob/develop/R/utilities-steady-state.R

To create excel files that can be loaded in R or MoBi, and here is an example how to use the function during simulation initialization (run to steady state and set initial values):

https://github.com/esqLABS/esqlabsR/blob/797e39da24e6921044f17539314f181eb5608799/R/utilities-simulation.R#L80-L93

**Comment by teutonicod** [comment]
Hi @PavelBal, 

I would also like to get a molecule equilibrated in MoBi and I tried to use the functions you mentioned before but I did not managed to get all the needed function installed, in order to use the equilibration functions. 

Could you please indicate to me which package or functions are needed to run the function `exportSteadyStateToXLS` ?
According to my understanding, this function is the closest one to what I would like to do (get the equilibrated parameters and import them in MoBi).

Thanks a lot for your help!

Donato

**Comment by PavelBal** [comment]
@teutonicod Install the `esqlabsR` package as described here: https://github.com/esqLABS/esqlabsR#installation
The function you need is [`exportSteadyStateToXLS`](https://esqlabs.github.io/esqlabsR/dev/reference/exportSteadyStateToXLS.html)

**Reply by teutonicod** [reply]
@PavelBal I already tried, but the package does not get installed, I think the problem is since I do not have Git installed. Is there a way to download the .zip of the package? (I didn't find it on the repo). Otherwise, I have downloaded the repository on my computer, is thee a way to load it from there?
Thanks a lot for your help!

**Reply by PavelBal** [reply]
Hm yes, the preferred installation requires Git. You can download the package here https://ci.appveyor.com/api/buildjobs/3h8bfwkesb4fs63x/artifacts/esqlabsR.zip

It might require installation of additional packages from CRAN then.

**Reply by teutonicod** [reply]
Thanks a lot, I tried to install all the packages manually but I finally when loading the esqlabsR package I get an error for a cyclic package dependencies, between esqlabsR and ospsuite.parameteridentification. I tried to modify the namespace file in the package but did not manage to solve the error.

**Reply by teutonicod** [reply]
I think the error is because the two packages ave a dependence for each-other in their namespace

**Reply by msevestre** [reply]
ospsuite.parameteridentification should not bed referencing esqlabsR

**Comment by PavelBal** [comment]
[ospsuite.parameteridentification_1.2.8.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/11817755/ospsuite.parameteridentification_1.2.8.zip)

Please use this version of `ospsuite.parameteridentification`.

**Reply by teutonicod** [reply]
Thanks a lot, with this new version worked. Before I used the only one available on the repository (v1.1)
https://github.com/Open-Systems-Pharmacology/OSPSuite.ParameterIdentification/tags

Thanks a lot, now it works!
